dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -952,102 +1078,143 @@ private void setStructureDependencies() throws DotDataException, DotSecurityExce
+							.getStructureByInode(relationship.getParentStructureInode());
+					solvedContentTypes.add(stInode);
+					contentTypes.addOrClean(relationship.getParentStructureInode(),
+							struct.getModDate());
+
+					if (!solvedContentTypes.contains(relationship.getParentStructureInode()))
+						structureDependencyHelper(relationship.getParentStructureInode(),
+								publisherFilter);
+				}
 			}
 		}
 	}
 
 	/**
 	 * 
-	 * @param cons
+	 * @param contentletSet
 	 * @throws DotDataException
 	 * @throws DotSecurityException
 	 */
-	private void processList(Set<Contentlet> cons) throws DotDataException, DotSecurityException {
-		Set<Contentlet> contentsToProcess = new HashSet<Contentlet>();
-		Set<Contentlet> contentsWithDependenciesToProcess = new HashSet<Contentlet>();
-
-		//Getting all related content
-
-		for (Contentlet con : cons) {
-			Host h = APILocator.getHostAPI().find(con.getHost(), user, false);
-			hosts.addOrClean( con.getHost(), h.getModDate()); // add the host dependency
-			contentsToProcess.add(con);
-
-			Map<Relationship, List<Contentlet>> contentRel =
-					APILocator.getContentletAPI().findContentRelationships(con, user);
-
-			for (Relationship rel : contentRel.keySet()) {
-				contentsToProcess.addAll(contentRel.get(rel));
-				/**
-				 * ISSUE #2222: https://github.com/dotCMS/dotCMS/issues/2222
-				 *
-				 * We need the relationships in which the single related content is involved.
-				 *
-				 */
-				if(contentRel.get(rel).size()>0)
-					relationships.addOrClean( rel.getInode(), rel.getModDate());
+	private void processList(final Set<Contentlet> contentletSet, final PublisherFilter publisherFilter) throws DotDataException, DotSecurityException {
+		final Set<Contentlet> contentsToProcess = new HashSet<Contentlet>();
+		final Set<Contentlet> contentsWithDependenciesToProcess = new HashSet<Contentlet>();
+
+		for (final Contentlet contentlet : contentletSet) {
+			// Host Dependency
+			if(!publisherFilter.doesExcludeDependencyClassesContainsType(PusheableAsset.SITE.getType())) {
+				final Host host = APILocator.getHostAPI().find(contentlet.getHost(), user, false);
+				hosts.addOrClean(contentlet.getHost(), host.getModDate());
+			}
+
+			contentsToProcess.add(contentlet);
+
+			// Relationships Dependencies
+			if(publisherFilter.isRelationships()) {
+				final Map<Relationship, List<Contentlet>> contentRelationshipsMap =
+						APILocator.getContentletAPI().findContentRelationships(contentlet, user);
+
+				for (final Relationship relationship : contentRelationshipsMap.keySet()) {
+					contentsToProcess.addAll(contentRelationshipsMap.get(relationship));
+					/**
+					 * ISSUE #2222: https://github.com/dotCMS/dotCMS/issues/2222
+					 *
+					 * We need the relationships in which the single related content is involved.
+					 *
+					 */
+					if (contentRelationshipsMap.get(relationship).size() > 0)
+						relationships.addOrClean(relationship.getInode(), relationship.getModDate());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)